includes/class-wc-fgc-update-variation-cart.php

```diff
@@ -0,0 +1,387 @@
+				'flexslider_enabled'          => get_theme_support( 'wc-product-gallery-slider' ),
+				'server_error'                => __( '🤕 Sorry, an error occured, please try again later.', 'wc_free_gift_coupons'),
+			);
+
+			// Enqueue needed css for it.
+			wp_enqueue_style( self::$name . '_style', plugin_dir_url( _WC_FGC_PLUGIN_NAME ) . 'assets/css/frontend-variation-cart' . $min_ext . '.css', array(), self::$version, 'all' );
+
+			// localize array here.
+			wp_localize_script( self::$name . '_js', 'wc_fgc_var_cart_params', $wc_fgc_trans_array );
+		}
+
+	}
+
+	/**
+	 * Change alert text for Variation.
+	 * 
+	 * Adjust selection alert text for variation popup.
+	 * 
+	 * @hook woocommerce_get_script_data
+	 * @param array $params
+	 * @param string $handle
+	 * @return $array
+	 */
+	public static function change_variation_select_alert( $params, $handle ) {
+		if ('wc-add-to-cart-variation' === $handle ) {
+			$params['i18n_make_a_selection_text'] = __( '😐 Please select some product options.', 'wc_free_gift_coupons' );
+		}
+		return $params;
+	}
+
+	/**
+	 * Add data on the product page.
+	 *
+	 * @name variation_update_data
+	 * @since 1.0.0
+	 * @param array        $cart_item  Cart item array.
+	 * @param array string $cart_item_key Cart item key.
+	 */
+	public static function variation_update_data( $cart_item, $cart_item_key ) {
+
+		$_product = $cart_item['data'];
+
+		$product_id = isset( $cart_item['product_id'] ) ? intval( $cart_item['product_id'] ) : 0; // Get the variation id.
+		$variation_id = isset( $cart_item['variation_id'] ) ? intval( $cart_item['variation_id'] ) : 0; // Get the variation id.
+	
+		$get_gift_cart_meta = isset( $cart_item['free_gift'] ) ? $cart_item['free_gift'] : '';
+
+		// Check if product is varaible and has free gift item meta in it.
+		if ( ( $_product->is_type( 'variable' ) || $variation_id > 0 ) && ! empty( $get_gift_cart_meta ) ) {
+
+			// Check if no variation selected, so as to do something cool :)
+			if ( $variation_id > 0 ) {
+				$edit_in_cart_text = _x( 'Change options', 'edit in cart link text', 'wc_free_gift_coupons' );
+				$var_edit_trigger_class = '';
+			}
+			else {
+				$edit_in_cart_text = _x( 'Choose options', 'edit in cart link text', 'wc_free_gift_coupons' );
+				$var_edit_trigger_class = ' wc-fgc-show-edit';
+			}
+
+			// Translators: %1$s text for edit price link.
+			$edit_in_cart_link_content = sprintf( __( '<small>%1$s<span class="dashicons dashicons-after dashicons-edit"></span></small>', 'edit in cart text', 'wc_free_gift_coupons' ), $edit_in_cart_text );
+
+			$variation_html =
+			'<div id="wc-fgc-item_' . esc_attr( $cart_item_key ) . '" class="actions wc-fgc-cart-update' . esc_attr( $var_edit_trigger_class ) . '">
+				<a href="javascript:void(0)" class="wc-fgc-edit-var-link button wc_fgc_updatenow" data-item_key="'. esc_attr( $cart_item_key ) .'" data-product_id="' . esc_attr( $product_id ) . '" data-variation_id="' . esc_attr( $variation_id ) . '">'
```

Comment:
Do we need `javascript:void(0)` or will `e.preventDefault()` do in the script itself? And if there's no href.... it doesn't degrade if there's no JS. The original version of this redirected back to the single product page, can we leave that and progressively enhance with edit in cart?

Reply:
Okay then.